use DataArray.assign_coords, Dataset.assign_coords or Dataset.assign as appropriate (new xarray requirements)

https://travis-ci.com/github/corteva/rioxarray/jobs/310700253
```
open_method = functools.partial(<function open_rasterio at 0x123f4a5f0>, masked=True)

windowed = True

tmpdir = local('/private/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/pytest-of-travis/pytest-0/test_to_raster_3d_open_method20')

    @pytest.mark.parametrize(

        "open_method, windowed",

        [

            (xarray.open_dataset, True),

            (xarray.open_dataset, False),

            (partial(rioxarray.open_rasterio, masked=True), True),

            (partial(rioxarray.open_rasterio, masked=True), False),

        ],

    )

    def test_to_raster_3d(open_method, windowed, tmpdir):

        tmp_raster = tmpdir.join("planet_3d_raster.tif")

>       with open_method(os.path.join(TEST_INPUT_DATA_DIR, "PLANET_SCOPE_3D.nc")) as mda:

test/integration/test_integration_rioxarray.py:796: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

rioxarray/_io.py:674: in open_rasterio

    mask_and_scale=mask_and_scale,

rioxarray/_io.py:494: in _load_subdatasets

    default_name=subdataset.split(":")[-1].lstrip("/").replace("/", "_"),

rioxarray/_io.py:746: in open_rasterio

    _decode_datetime_cf(result)

rioxarray/_io.py:442: in _decode_datetime_cf

    data_array[coord].values = new_values

../../../miniconda/envs/test/lib/python3.7/site-packages/xarray/core/common.py:262: in __setattr__

    object.__setattr__(self, name, value)

../../../miniconda/envs/test/lib/python3.7/site-packages/xarray/core/dataarray.py:562: in values

    self.variable.values = value

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <xarray.IndexVariable 'time' (time: 2)>

array([     0.      , 872712.659688])

Attributes:

    _FillValue:  nan

values = array([cftime.DatetimeProlepticGregorian(2016-12-19 10:27:29),

       cftime.DatetimeProlepticGregorian(2016-12-29 12:52:41.659696)],

      dtype=object)

    @Variable.values.setter  # type: ignore

    def values(self, values):

        raise ValueError(

>           f"Cannot assign to the .values attribute of dimension coordinate a.k.a IndexVariable {self.name!r}. "

            f"Please use DataArray.assign_coords, Dataset.assign_coords or Dataset.assign as appropriate."

        )

E       ValueError: Cannot assign to the .values attribute of dimension coordinate a.k.a IndexVariable 'time'. Please use DataArray.assign_coords, Dataset.assign_coords or Dataset.assign as appropriate.
```

--- a/docs/history.rst
+++ b/docs/history.rst
@@ -4,6 +4,7 @@ History
 0.0.24
 ------
 - ENH: Add variable names to error messages for clarity (pull #99)
+- BUG: Use assign_coords in _decode_datetime_cf (issue #101)
 
 0.0.23
 ------

--- a/rioxarray/_io.py
+++ b/rioxarray/_io.py
@@ -416,8 +416,17 @@ def _decode_datetime_cf(data_array):
             new_values = times.decode_cf_timedelta(
                 data_array[coord].values, units=units
             )
-            new_values = new_values.astype(np.dtype("timedelta64[ns]"))
-            data_array[coord].values = new_values
+            data_array = data_array.assign_coords(
+                {
+                    coord: IndexVariable(
+                        dims=data_array[coord].dims,
+                        data=new_values.astype(np.dtype("timedelta64[ns]")),
+                        attrs=data_array[coord].attrs,
+                        encoding=data_array[coord].encoding,
+                    )
+                }
+            )
+
         # stage 2: datetime
         if (
             "units" in data_array[coord].attrs
@@ -438,8 +447,17 @@ def _decode_datetime_cf(data_array):
                 calendar=calendar,
                 use_cftime=True,
             )
-            new_values.astype(dtype)
-            data_array[coord].values = new_values
+            data_array = data_array.assign_coords(
+                {
+                    coord: IndexVariable(
+                        dims=data_array[coord].dims,
+                        data=new_values.astype(dtype),
+                        attrs=data_array[coord].attrs,
+                        encoding=data_array[coord].encoding,
+                    )
+                }
+            )
+    return data_array
 
 
 def _parse_driver_tags(riods, attrs, coords):
@@ -743,7 +761,7 @@ def open_rasterio(
 
     # update attributes from NetCDF attributess
     _load_netcdf_attrs(riods.tags(), result)
-    _decode_datetime_cf(result)
+    result = _decode_datetime_cf(result)
 
     # make sure the _FillValue is correct dtype
     if "_FillValue" in attrs:

